File: aegis-backend/app/models/security_event.py
from sqlalchemy import ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel


class SecurityEvent(BaseModel):
    """Sanitized proxy outcome used for traffic and security analytics."""

    __tablename__ = "security_events"

    event_type: Mapped[str] = mapped_column(String(50), nullable=False, index=True)
    source_ip: Mapped[str] = mapped_column(String(45), nullable=False, index=True)
    api_key_id: Mapped[int | None] = mapped_column(
        ForeignKey("api_keys.id"), nullable=True, index=True
    )
    rule_id: Mapped[int | None] = mapped_column(Integer, nullable=True)
    method: Mapped[str] = mapped_column(String(10), nullable=False)
    path: Mapped[str] = mapped_column(String(2048), nullable=False)
    status_code: Mapped[int] = mapped_column(Integer, nullable=False, index=True)

File: reverse-proxy/internal/proxy/security_event.go
package proxy

type SecurityEvent struct {
	EventType  string `json:"event_type"`
	SourceIP   string `json:"source_ip"`
	APIKeyID   *int64 `json:"api_key_id"`
	RuleID     *int64 `json:"rule_id"`
	Method     string `json:"method"`
	Path       string `json:"path"`
	StatusCode int    `json:"status_code"`
}

type EventReporter interface {
	Report(event SecurityEvent)
}
